network.go

```diff
@@ -1184,6 +1290,31 @@ func isPhysicalIface(ifaceName string) (bool, error) {
 	return true, nil
 }
 
+// Long term, this should be made more configurable.  For now matching path
+// provided by CNM VPP plugin
+const hostSocketSearchPath = "/tmp/vhostuser_%s/vhu.sock"
+
+// isVhostuserIface checks if an interface is a dummy placeholder
+// for a vhost-user socket
+func isVhostuserIface(netInfo NetworkInfo) (bool, string, error) {
+	if netInfo.Iface.Name == "lo" {
+		return false, "", nil
+	}
+
+	// for now, check for socket file existence at known location.
+	// Identify this be searching for a file in a path which contains
+	// the interfaces IP address.  ie:
+	//  /tmp/vhostuser_<ipv4 address>/vhu.sock
+	for _, addr := range netInfo.Addrs {
+		socketPath := fmt.Sprintf(hostSocketSearchPath, addr.IPNet.IP)
```

Comment:
It would make sense to have a `vhosUserSocketPath(info interface{}) (string, error)` function that would try all known heuristics (one for now) for finding a  vhost-user socket path and return it if it can find one.

Reply:
Sure. @sameo  - you mean to say add a function to the vhostUserDevice interface, right?  Good call...

Reply:
@egernst yes, `isVhostUserInterface()` is almost what I'm looking for, except that:

1. It's network specific
2. It's `clearcontainers/vpp` plugin specific
3. It returns both a boolean and the socket string

So my suggestion is to make it generic, have it embed the `clearcontainers/vpp` specific bits, and only returns a `(string, error)` where a nil string means no socket path could be found.
Once we understand how to handle the storage case, we can just add it to this routine.